core/templates/components/question-directives/questions-list/questions-list.component.spec.ts

```diff
@@ -996,13 +996,31 @@ describe('QuestionsListComponent', () => {
     expect(ctrl.saveAndPublishQuestion).toHaveBeenCalledWith('commit');
   });
 
+  it('should update skill linkage correctly', () => {
+    ctrl.skillLinkageModificationsArray = [
+      {
+        id: 'skillId1',
+        task: 'update_difficulty',
+        difficulty: 0.9
+      }
+    ];
+    spyOn(EditableQuestionBackendApiService, 'editQuestionSkillLinksAsync')
+      .and.returnValue($q.resolve());
+
+    ctrl.updateSkillLinkage();
+    $scope.$apply();
+    $timeout.flush(500);
```

Comment:
Is this sure to be non-flaky?

Reply:
This test is more or less a copy of the test above. That test already had the timeout statement in develop and it hasn't been flaky.